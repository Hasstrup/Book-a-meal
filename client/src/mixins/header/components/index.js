import React from 'react';
import { LogOutUser } from '../../../actions/users/'

/**
   * @name LoggedOutHeader
   * @description This header would only be rendered when there is no user in session
   * @param {object} props the props containing the history object for navigation
   * @returns {function} React Component
   */
export const LoggedOutHeader = ({ history }) => (
  <header>
    <div className="header">
      <div className="home-tag">
        <p className="logo" onClick={() => { history.push('/'); }}> Book <br />
            a meal
        </p>
      </div>
      <div className="header-buttons">
        <p className="h1-login-button"> Login </p>
        <p> Sign Up</p>
        <p className="h1-login-button" onClick={() => { history.push('/catalogue'); }}> Catalogue </p>
      </div>
    </div>
  </header>
);


  /**
   * @name LoggedInHeader
   * @description This header would only be rendered when a user is logged in
   * @param {object} props the props containing the history object for navigation
   * @returns {function} React Component
   */
export const LoggedInHeader = ({ history }) => (
  <header>
    <div className="header">
      <div className="home-tag">
        <p className="logo" onClick={() => { history.push('/'); }}> Book <br />
            a meal
          </p>
      </div>
      <div className="header-buttons">
        <p className="h1-login-button" onClick={() => { history.push('/orders')}}> Cart & Orders </p>
        <p onClick={() => { history.push('/profile')}} > Workstation </p>
        <p className="h1-login-button" onClick={() => { history.push('/catalogue')}}> Catalogue </p>
        <p className="h1-login-button" onClick={() => { LogOutUser(history) }}> Logout </p>
      </div>
    </div>
  </header>
);
