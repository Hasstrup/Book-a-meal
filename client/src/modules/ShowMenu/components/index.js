import ShowMeal from './ShowMeal';
import ShowMenuDetails from './ShowMenuDetails';
import VendorDetails from './VendorDetails';

export const ComponentList = [ShowMeal, ShowMenuDetails, VendorDetails];
export const PropList = [{}, { meals: [] }, { menu: { meals: [], description: '', name: '', kitchen: { image: '', name: '', description: '' }}}];
export const NameList = ['Show Meal', 'ShowMenuDetails', 'VendorDetails'];

