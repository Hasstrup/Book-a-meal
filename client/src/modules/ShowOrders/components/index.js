import CurrentOrder from './CurrentOrder';
import OrderHistoryGrid from './OrderHistoryGrid';
import { OrderHistoryComponent as RenderMainHistory } from './RenderMainHistory';

export const ComponentList = [
  CurrentOrder,
  OrderHistoryGrid,
  RenderMainHistory
];

export const PropList = [
  { meals: [], history: () => {} },
  { orders: [], title: '' },
  { orders: [] }
];

export const NameList = [
  'Current Order',
  'OrderHistoryGrid',
  'RenderMainHistory'
];
