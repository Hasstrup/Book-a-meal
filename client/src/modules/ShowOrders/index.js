import React, { Component } from 'react';
import { connect } from 'react-redux';
import swal from 'sweetalert';
import CurrentOrderComponent from './components/CurrentOrder';
import RenderActualHistory from './components/RenderMainHistory';
import OrderHistoryHeader from './components/OrderHistoryHeader';
import AuthOnly from '../../hocs/AuthOnly';
import { DispatchNotification } from '../../actionTypes/misc';
import { fetchAllOrders } from '../../actions/orders';
import { CartOps } from '../../actions/ops/cart';
import '../styles/orders.scss';


class OrderHistoryPage extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.state.target = this.props.user;
    this.state.ordersInView = this.props.orders[this.props.user.id];
    this.state.currentMealsInCart = CartOps()()({});
  }

    componentDidMount = () => {
      const { dispatch, user, kitchen } = this.props;
      /* eslint-disable-next-line  */
      user && dispatch(fetchAllOrders('user')(user.id));
      kitchen && dispatch(fetchAllOrders('kitchen')(kitchen.id));
    }

    componentDidUpdate = (prevProps) => {
      const { orders } = this.props;
      const { target } = this.state;
      if (orders[target.id] && orders[target.id].length) {
        if (!prevProps.orders[target.id] || !prevProps.orders[target.id].length) return this.setState({ ordersInView: this.props.orders[this.state.target.id] });
      }
    }

    changeTarget = (target) => {
      const { dispatch, orders } = this.props;
      dispatch(DispatchNotification(`You are now seeing orders made ${target.name ? 'to' : 'by'} ${target.name || target.firstname}`));
      dispatch(fetchAllOrders(target.type)(target.id));
      this.state.target = target;
      // this rerender here will update the dom;
      this.setState({ ordersInView: orders[`${target.id}`] });
    }

    render = () => (
      <div className="main-body-component" style={{ minHeight: '100vh' }}>
        <OrderHistoryHeader kitchen={this.props.kitchen} handleChange={this.changeTarget} user={this.props.user} target={this.state.target} />
        <div className="orders-page-side-one-main">
          { this.state.currentMealsInCart && this.state.currentMealsInCart.length && <CurrentOrderComponent meals={this.state.currentMealsInCart} history={this.props.history} /> }
          <RenderActualHistory
            orders={this.state.ordersInView}
            determineRenderBy={this.state.ordersInView}
            callback={() => {
            this.props.history.push('/catalogue');
            swal(`Here you go ${this.props.user.firstname}, make your from here`, 'PS: they are all yummy', 'success');
          }}
            text={this.state.target.id === this.props.user.id ? ` Aw schucks, ${this.state.target.firstname} hasn't made an order yet` : 'Looks like no one has made an order yet'}
            subtitle="order your first item"
            emptyContainerStyle={{
            height: '10%',
            position: 'relative',
            bottom: '-14vh'
          }}
          />
        </div>
      </div>
    )
}

const mapStateToProps = state => ({
  kitchen: state.kitchens.target,
  user: state.users.current,
  orders: state.orders.allOrders || {}
});

export default connect(mapStateToProps)(AuthOnly(OrderHistoryPage));
