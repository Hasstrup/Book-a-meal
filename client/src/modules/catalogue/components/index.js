import { CatalogueFirstRow, CatalogueGridMainX } from '../index';

export const ComponentList = [CatalogueFirstRow, CatalogueGridMainX];
export const PropList = [{}, { menus: [] }];
export const NameList = ['Catalogue Preamble', 'Catalogue Grid'];

