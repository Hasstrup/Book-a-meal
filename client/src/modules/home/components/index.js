import AuthComponent from './auth';
import BannerComponent from './banner';
import FloatingComponent from './floater';

export const ComponentList = [AuthComponent, BannerComponent, FloatingComponent];
export const PropList = [{ logInUser: () => {}, createUser: () => {} }, {}, {}];
export const NameList = ['Auth Component', 'Banner Component', 'Floating Component'];