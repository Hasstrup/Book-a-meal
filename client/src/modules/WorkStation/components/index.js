import RenderMeals from './meals';
import RenderOrdersComponent from './orders';
import UserBio from './UserBio';
import { WorkStation as WorkStationMain } from './workstation-main';
import { MenuOfTheDayContainer as MenuOfTheDay } from './workstation-main/modules/MenuOfTheDay';


export const ComponentList =
[
  RenderMeals,
  RenderOrdersComponent,
  UserBio,
  WorkStationMain,
  MenuOfTheDay
];

export const PropList = [
  { handleSubmit: () => {}, meals: [] },
  {},
  {
    user: { firstname: '', email: '', bio: '' },
    handleChange: () => {},
    handleSubmit: () => {}
  },
  {
    kitchen: { name: '', description: '', bio: '' },
    meals: [],
    handleSubmit: () => {}
  },
  {
    dispatch: () => {},
    ofTheDay: { name: '', meals: [], description: '' },
    meals: []
  }

];
export const NameList = [
  'RenderMeals',
  'RenderOrdersComponent',
  'User Bio',
  'WorkStation Main',
  'MenuOfTheDayContainer'
];

