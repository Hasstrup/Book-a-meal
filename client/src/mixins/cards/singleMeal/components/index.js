import React from 'react';
import { changePhoto, HideMealCard } from '../../../../modules/WorkStation/utils';
import { AddToCart } from '../../../../actions/orders/';


const defaultUrl = 'https://images.unsplash.com/photo-1453831362806-3d5577f014a4?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=9dd8da96be0724ab84e4147d428f6bba&auto=format&fit=crop&w=500&q=60';
/**
 * @name SingleMealCard
 * @returns {function} React Component
 * @param {Object} Props The props passed to the item
 * wshould contain the current kitchen the item and the item;
 * @description Single Meal Card component, recieves data from it's parent container
 */
const SingleMealCard = ({ meal, kitchen, handleDelete, handleEdit, displayText, wantsToEdit, renderEditable, dispatch }) => (
    <div className="display-menu-item" id={meal.id} onMouseLeave={() => { HideMealCard(meal.id) }}>
      <div className="order-add-item">
        <img src={meal.image || defaultUrl} alt={meal.name} id={`image-${meal.id}`} className="main-meal-item-img" />
        <div className={`change-preview-image modifiable${meal.id}`} style={{ opacity: wantsToEdit ? 1 : 0, zIndex: wantsToEdit ? 5 : -1, display: wantsToEdit ? 'block' : 'none' }}>
          <p className="label-for-input"> Click to change Image </p>
          <input type="file" className={`input-file image-${meal.id}`} name="input-file" onChange={() => { changePhoto(`image-${meal.id}`)}} accept=".jpg, .png, .jpeg" />
        </div>
      </div>
      <div className="item-details">
        <div className="item-title-and-description">
          <p className={`item-title-main editable editable${meal.id}`}> {meal.name || ''} </p>
          <p className={`item-description editable editable${meal.id}`}> {meal.description || ''} </p>
        </div>
        <div className="buttons-array-and-togglers">
          {
          kitchen && kitchen.id === meal.kitchenId ?
          <p className="option-options"> { renderEditable && <span className="edit-meal-option" onClick={handleEdit} id={`target${meal.id}`}> {displayText} </span> } { wantsToEdit ? null : <span className="delete-meal-option" onClick={handleDelete}> Delete </span> } </p>
          : <p className="option-options" onClick={() => dispatch(AddToCart()(meal))}> Order </p>
        }
  
          <p className="option-options2" > N{meal.price} </p>
        </div>
      </div>
    </div>
);

export default SingleMealCard;