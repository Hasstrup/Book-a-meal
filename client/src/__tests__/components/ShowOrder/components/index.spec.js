import TestComponent from '../../../TestComponents';
import { ComponentList, PropList, NameList } from '../../../../modules/ShowOrders/components';

(async () => {
  await TestComponent(ComponentList, NameList, PropList, 'Show Orders Components');
})();