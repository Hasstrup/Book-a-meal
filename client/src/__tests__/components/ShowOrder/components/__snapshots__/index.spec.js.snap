// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Orders Components Current Order should render correctly on shallow render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    history={[Function]}
    meals={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="display-current-order-heading"
        >
          <p
            className="display-current-order-heading-head"
          >
             Current Order 
          </p>
          <p
            className="display-current-order-heading-desc"
          >
            This contains all the meal options you have placed orders for in this current session.
          </p>
        </div>,
        <Connect(DisplayOrderCard)
          editable={true}
          history={[Function]}
          meals={Array []}
        />,
      ],
      "className": "display-current-order",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="display-current-order-heading-head"
            >
               Current Order 
            </p>,
            <p
              className="display-current-order-heading-desc"
            >
              This contains all the meal options you have placed orders for in this current session.
            </p>,
          ],
          "className": "display-current-order-heading",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Current Order ",
              "className": "display-current-order-heading-head",
            },
            "ref": null,
            "rendered": " Current Order ",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This contains all the meal options you have placed orders for in this current session.",
              "className": "display-current-order-heading-desc",
            },
            "ref": null,
            "rendered": "This contains all the meal options you have placed orders for in this current session.",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "editable": true,
          "history": [Function],
          "meals": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="display-current-order-heading"
          >
            <p
              className="display-current-order-heading-head"
            >
               Current Order 
            </p>
            <p
              className="display-current-order-heading-desc"
            >
              This contains all the meal options you have placed orders for in this current session.
            </p>
          </div>,
          <Connect(DisplayOrderCard)
            editable={true}
            history={[Function]}
            meals={Array []}
          />,
        ],
        "className": "display-current-order",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="display-current-order-heading-head"
              >
                 Current Order 
              </p>,
              <p
                className="display-current-order-heading-desc"
              >
                This contains all the meal options you have placed orders for in this current session.
              </p>,
            ],
            "className": "display-current-order-heading",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Current Order ",
                "className": "display-current-order-heading-head",
              },
              "ref": null,
              "rendered": " Current Order ",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This contains all the meal options you have placed orders for in this current session.",
                "className": "display-current-order-heading-desc",
              },
              "ref": null,
              "rendered": "This contains all the meal options you have placed orders for in this current session.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "editable": true,
            "history": [Function],
            "meals": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Show Orders Components OrderHistoryGrid should render correctly on shallow render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    orders={Array []}
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="today-text"
        >
          <div
            className="today-introduction"
          >
            <p
              className="today-intro"
            >
               
              
               
            </p>
            <p
              className="today-intro-desc"
            >
              This is a collection of all the batches of items you paid for today. Had dinner yet?
            </p>
          </div>
        </div>,
        <div
          className="order-story-grid"
        />,
      ],
      "className": "today-story",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="today-introduction"
          >
            <p
              className="today-intro"
            >
               
              
               
            </p>
            <p
              className="today-intro-desc"
            >
              This is a collection of all the batches of items you paid for today. Had dinner yet?
            </p>
          </div>,
          "className": "today-text",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="today-intro"
              >
                 
                
                 
              </p>,
              <p
                className="today-intro-desc"
              >
                This is a collection of all the batches of items you paid for today. Had dinner yet?
              </p>,
            ],
            "className": "today-introduction",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "",
                  " ",
                ],
                "className": "today-intro",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "",
                " ",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "This is a collection of all the batches of items you paid for today. Had dinner yet?",
                "className": "today-intro-desc",
              },
              "ref": null,
              "rendered": "This is a collection of all the batches of items you paid for today. Had dinner yet?",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "order-story-grid",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="today-text"
          >
            <div
              className="today-introduction"
            >
              <p
                className="today-intro"
              >
                 
                
                 
              </p>
              <p
                className="today-intro-desc"
              >
                This is a collection of all the batches of items you paid for today. Had dinner yet?
              </p>
            </div>
          </div>,
          <div
            className="order-story-grid"
          />,
        ],
        "className": "today-story",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="today-introduction"
            >
              <p
                className="today-intro"
              >
                 
                
                 
              </p>
              <p
                className="today-intro-desc"
              >
                This is a collection of all the batches of items you paid for today. Had dinner yet?
              </p>
            </div>,
            "className": "today-text",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="today-intro"
                >
                   
                  
                   
                </p>,
                <p
                  className="today-intro-desc"
                >
                  This is a collection of all the batches of items you paid for today. Had dinner yet?
                </p>,
              ],
              "className": "today-introduction",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "",
                    " ",
                  ],
                  "className": "today-intro",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "",
                  " ",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "This is a collection of all the batches of items you paid for today. Had dinner yet?",
                  "className": "today-intro-desc",
                },
                "ref": null,
                "rendered": "This is a collection of all the batches of items you paid for today. Had dinner yet?",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "order-story-grid",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Show Orders Components RenderMainHistory should render correctly on shallow render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderHistoryComponent
    orders={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="introduce-history"
        >
          <p
            className="introduce-history-main"
          >
             History 
          </p>
          <p
            className="introduce-history-desc"
          >
            So we've been trying to help you enjoy the experience on this service, that's why we took the time to stack all your orders together, so you have some insights on things you like. Enjoy
          </p>
        </div>,
        <div
          className="main-history-storybook"
        />,
      ],
      "className": "history-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="introduce-history-main"
            >
               History 
            </p>,
            <p
              className="introduce-history-desc"
            >
              So we've been trying to help you enjoy the experience on this service, that's why we took the time to stack all your orders together, so you have some insights on things you like. Enjoy
            </p>,
          ],
          "className": "introduce-history",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " History ",
              "className": "introduce-history-main",
            },
            "ref": null,
            "rendered": " History ",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "So we've been trying to help you enjoy the experience on this service, that's why we took the time to stack all your orders together, so you have some insights on things you like. Enjoy",
              "className": "introduce-history-desc",
            },
            "ref": null,
            "rendered": "So we've been trying to help you enjoy the experience on this service, that's why we took the time to stack all your orders together, so you have some insights on things you like. Enjoy",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            null,
          ],
          "className": "main-history-storybook",
        },
        "ref": null,
        "rendered": Array [
          null,
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="introduce-history"
          >
            <p
              className="introduce-history-main"
            >
               History 
            </p>
            <p
              className="introduce-history-desc"
            >
              So we've been trying to help you enjoy the experience on this service, that's why we took the time to stack all your orders together, so you have some insights on things you like. Enjoy
            </p>
          </div>,
          <div
            className="main-history-storybook"
          />,
        ],
        "className": "history-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="introduce-history-main"
              >
                 History 
              </p>,
              <p
                className="introduce-history-desc"
              >
                So we've been trying to help you enjoy the experience on this service, that's why we took the time to stack all your orders together, so you have some insights on things you like. Enjoy
              </p>,
            ],
            "className": "introduce-history",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " History ",
                "className": "introduce-history-main",
              },
              "ref": null,
              "rendered": " History ",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "So we've been trying to help you enjoy the experience on this service, that's why we took the time to stack all your orders together, so you have some insights on things you like. Enjoy",
                "className": "introduce-history-desc",
              },
              "ref": null,
              "rendered": "So we've been trying to help you enjoy the experience on this service, that's why we took the time to stack all your orders together, so you have some insights on things you like. Enjoy",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              null,
              null,
            ],
            "className": "main-history-storybook",
          },
          "ref": null,
          "rendered": Array [
            null,
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
